File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/aspplus/samples/vstudio/services/datatypes/vb/DataTypes.asmx.vb
'-----------------------------------------------------------------------
'  This file is part of the Microsoft .NET SDK Code Samples.
' 
'  Copyright (C) Microsoft Corporation.  All rights reserved.
' 
'This source code is intended only as a supplement to Microsoft
'Development Tools and/or on-line documentation.  See these other
'materials for detailed information regarding Microsoft code samples.
' 
'THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
'KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
'IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
'PARTICULAR PURPOSE.
'-----------------------------------------------------------------------

Imports System.Web.Services
Imports System.Reflection

< _
    Assembly : AssemblyTitle(""), _
    Assembly : AssemblyDescription("A QuickStart Tutorial Assembly"), _
    Assembly : AssemblyConfiguration(""), _
    Assembly : AssemblyCompany("Microsoft Corporation"), _
    Assembly : AssemblyProduct("Microsoft QuickStart Tutorials"), _
    Assembly : AssemblyCopyright(" Microsoft Corporation.  All rights reserved."), _
    Assembly : AssemblyTrademark(""), _
    Assembly : AssemblyCulture(""), _
    Assembly : AssemblyVersion("1.1.*"), _
    Assembly : AssemblyDelaySign(false), _
    Assembly : AssemblyKeyFile(""), _
    Assembly : AssemblyKeyName("") _
>

Namespace DataTypes.Vb
    Public Class DataTypesService
        Inherits System.Web.Services.WebService

#Region " XML Web services Designer Generated Code "

        'Required by the WebServices Designer
        Private components As System.ComponentModel.Container

        Public Sub New()
            MyBase.New()

            'CODEGEN: This procedure is required by the WebServices Designer
            'Do not modify it using the code editor.
            InitializeComponent()

            'Add your own initialization code after the InitializeComponent call
        End Sub

        Private Sub InitializeComponent()
            'CODEGEN: This procedure is required by the WebServices Designer
            'Do not modify it using the code editor.
            components = New System.ComponentModel.Container()
        End Sub

        Protected Overloads Overrides Sub Dispose(ByVal disposing as Boolean)
            'CODEGEN: This procedure is required by the WebServices Designer
            'Do not modify it using the code editor.
        End Sub

#End Region


        <WebMethod()> Public Function SayHello() As String

            Return "Hello World!"
        End Function

        <WebMethod()> Public Function SayHelloName(ByVal Name As String) As String

            Return "Hello " & Name
        End Function

        <WebMethod()> Public Function GetIntArray() As Integer()

            Dim I As Integer
            Dim A(5) As Integer
            For I = 0 To 4
                A(I) = I * 10
            Next
            Return A
        End Function

        <WebMethod()> Public Function GetMode() As Mode

            Return Mode.EOff
        End Function

        <WebMethod()> Public Function GetOrder() As Order

            Dim MyOrder As New Order()

            MyOrder.Price = 34.5
            MyOrder.OrderID = 323232

            Return MyOrder
        End Function

        <WebMethod()> Public Function GetOrders() As Order()

            Dim MyOrder(2) As Order

            MyOrder(0) = New Order()
            MyOrder(0).Price = 34.5
            MyOrder(0).OrderID = 323232
            MyOrder(1) = New Order()
            MyOrder(1).Price = 99.4
            MyOrder(1).OrderID = 645645

            Return MyOrder
        End Function


    End Class

    Public Enum Mode

        EOn = 1
        EOff = 2

    End Enum

    Public Class Order

        Public OrderID As Integer
        Public Price As Double
    End Class

End Namespace

File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/aspplus/samples/vstudio/services/datatypes/cs/DataTypes.asmx.cs
//-----------------------------------------------------------------------
//  This file is part of the Microsoft .NET SDK Code Samples.
// 
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//This source code is intended only as a supplement to Microsoft
//Development Tools and/or on-line documentation.  See these other
//materials for detailed information regarding Microsoft code samples.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//-----------------------------------------------------------------------

using System.Reflection;

[assembly: AssemblyTitle("")]
[assembly: AssemblyDescription("A QuickStart Tutorial Assembly")]
[assembly: AssemblyConfiguration("")]
[assembly: AssemblyCompany("Microsoft Corporation")]
[assembly: AssemblyProduct("Microsoft QuickStart Tutorials")]
[assembly: AssemblyCopyright(" Microsoft Corporation.  All rights reserved.")]
[assembly: AssemblyTrademark("")]
[assembly: AssemblyCulture("")]
[assembly: AssemblyVersion("1.1.*")]

namespace DataTypes.Cs
{
    using System;
    using System.Collections;
    using System.ComponentModel;
    using System.Diagnostics;
    using System.Web;
    using System.Web.Services;

    /// <summary>
    ///    Summary description for DataTypes.
    /// </summary>
    public class DataTypesService : System.Web.Services.WebService
    {
        public DataTypesService()
        {
            //CODEGEN: This call is required by the XML Web services created using ASP.NET Designer
            InitializeComponent();
        }

        /// <summary>
        ///    Required method for Designer support - do not modify
        ///    the contents of this method with the code editor.
        /// </summary>
        private void InitializeComponent()
        {
        }

        /// <summary>
        ///    Clean up any resources being used.
        /// </summary>
        protected override void Dispose(bool disposing)
        {
        }

		[WebMethod]
		public String SayHello()
		{
			return "Hello World!";
		}

		[WebMethod]
		public String SayHelloName(String name)
		{
			return "Hello " + name;
		}

		[WebMethod]
		public int[] GetIntArray()
		{
			int[] a = new int[5];
			for (int i=0; i<5; i++)
				a[i] = i*10;
			return a;
		}

		[WebMethod]
		public Mode GetMode()
		{
			return Mode.Off;
		}

		[WebMethod]
		public Order GetOrder()
		{
			Order myOrder = new Order();

			myOrder.Price=34.5;
			myOrder.OrderID = 323232;

			return myOrder;
		}

		[WebMethod]
		public Order[] GetOrders()
		{
			Order [] myOrder = new Order[2];

			myOrder[0] = new Order();
			myOrder[0].Price=34.5;
			myOrder[0].OrderID = 323232;
			myOrder[1] = new Order();
			myOrder[1].Price=99.4;
			myOrder[1].OrderID = 645645;

			return myOrder;
		}
	}

	public enum Mode
	{
		On = 1,
		Off = 2
	}

	public class Order
	{
		public int OrderID;
		public double Price;
	}

}
